var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -96,20 +83,20 @@ class Dealii(CMakePackage):
 
     # optional dependencies
     depends_on("mpi",              when="+mpi")
-    depends_on("arpack-ng+mpi",    when='+arpack+mpi')
```

Comment:
>  if I build +arpack~mpi it doesn't build with arpack, only when I build with +arpack+mpi? That doesn't seem... ideal

correct, that's the behaviour and it is indeed not ideal, but variant description clearly say so: `(only with MPI)`.

In this specific case of `arpack`, one actually can build (outside of Spack) `dealii+mpi^arpack~mpi`. Here it was only added for consistency. But other packages may get broken when you try this (it all depends).